dev/diffs/4.0.0.diff

```diff
@@ -159,6 +190,21 @@ index 698ca009b4f..57d774a3617 100644
  
  -- Test tables
  CREATE table  explain_temp1 (key int, val int) USING PARQUET;
+diff --git a/sql/core/src/test/resources/sql-tests/inputs/listagg-collations.sql b/sql/core/src/test/resources/sql-tests/inputs/listagg-collations.sql
+index aa3d02dc2fb..c4f878d9908 100644
+--- a/sql/core/src/test/resources/sql-tests/inputs/listagg-collations.sql
++++ b/sql/core/src/test/resources/sql-tests/inputs/listagg-collations.sql
+@@ -5,7 +5,9 @@ WITH t(c1) AS (SELECT listagg(col1) WITHIN GROUP (ORDER BY col1) FROM (VALUES ('
+ -- Test cases with utf8_lcase. Lower expression added for determinism
+ SELECT lower(listagg(c1) WITHIN GROUP (ORDER BY c1 COLLATE utf8_lcase)) FROM (VALUES ('a'), ('A'), ('b'), ('B')) AS t(c1);
+ WITH t(c1) AS (SELECT lower(listagg(DISTINCT col1 COLLATE utf8_lcase)) FROM (VALUES ('a'), ('A'), ('b'), ('B'))) SELECT len(c1), regexp_count(c1, 'a'), regexp_count(c1, 'b') FROM t;
+-SELECT lower(listagg(DISTINCT c1 COLLATE utf8_lcase) WITHIN GROUP (ORDER BY c1 COLLATE utf8_lcase)) FROM (VALUES ('a'), ('B'), ('b'), ('A')) AS t(c1);
++-- TODO https://github.com/apache/datafusion-comet/issues/1947
++-- TODO fix Comet for this query
++-- SELECT lower(listagg(DISTINCT c1 COLLATE utf8_lcase) WITHIN GROUP (ORDER BY c1 COLLATE utf8_lcase)) FROM (VALUES ('a'), ('B'), ('b'), ('A')) AS t(c1);
```

Comment:
Should we disable comet instead of commenting out? due to this, the result in `listagg-collations.sql.out` is also removed.

Reply:
Could we make that change as part of the follow-up issue https://github.com/apache/datafusion-comet/issues/1947 that is linked to here (or maybe we can just fix the test instead). The sooner we can get this PR merged, the easier it will be to start addressing the test failures in separate (and smaller) PRs.

I would also like to ship the experimental 4.0.0 support in Comet 0.9.0 so that we can start getting feedback from the community. It seems like all the skipped tests have links to issues now.